base/compiler/abstractinterpretation.jl

```diff
@@ -1334,7 +1346,7 @@ function typeinf_local(interp::AbstractInterpreter, frame::InferenceState)
                 pc´ = (stmt::GotoNode).label
             elseif isa(stmt, GotoIfNot)
                 condt = abstract_eval_value(interp, stmt.cond, s[pc], frame)
-                if condt === Bottom
+                if bail_out_local(interp, condt, frame)
```

Comment:
Does the `empty!(frame.pclimitations)` not depend on this being literal `Union{}`. How are you proposing to use this and what does that do to the `empty!` call?

Reply:
ah yes, it's not related to the bail out optimization and we should emptify `frame.pclimitations` regardless of whether we bail out or not.